### PR TITLE
add a "byte generation" script to the benchmarks

### PR DESCRIPTION
**Pull request by amtoine**
following a discussion with @fdncred on Discord.

# Description
```nushell
def "random bytes" [n: int]: nothing -> binary {
    seq 1 ($n / 8 + 1)
        | each { random integer }
        | into binary
        | enumerate
        | reduce -f 0x[] {|it, acc|
            $acc | bytes add $it.item
        }
        | first $n
}
```
and run it from `n = 10` to `n = 1_000_000`

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by amtoine** [comment]
i'm not sure how to make this work properly, all the other benchmarks parse file from the filesystem and `nu!` doesn't appear to work locally :cry:

File: benches/benchmarks.rs [hunk 1]
```diff
@@ -162,11 +163,40 @@ fn decoding_benchmarks(c: &mut Criterion) {
     group.finish();
 }
 
+fn generating_random_bytes(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Generating random bytes");
+
+    for n in [10, 100, 1_000, 10_000, 100_000, 1_000_000] {
```
Comment (sholderbach) [inline benches/benchmarks.rs:169]: With the whole overhead if our benchmark would be invoking new `nu` processes for each sample, I don't think there is much to glance from the low numbers. 

In general I am not sure how relevant it is to have a small scaling factor. Criterion will run the benchmark often enough to get stable data, so each scaling step will take a not insignificant time overall.

**Review by sholderbach — COMMENTED** [review]
In general I would be happy to have more criterion benchmarks, especially ones that exercise larger bits of the engine. But I am not sure if shelling out via `nu!` is a good way to do it. This just screams overhead. And if the benchmark runs are too voluminous to seldomly be run you don't gain anything. (profiling launched subprocesses is harder than profiling a single process)

We should certainly have some mixed command synthetic benchmarks, to get an overall sense if something changes due to engine changes. For a criterion suite of benchmarks optimized for profiling on the other hand we may want to limit the number of different commands to isolate the effects of different operations or access patterns.

For now `nu_scripts` is a great place to keep them around until we have a better setup.

**Comment by amtoine** [comment]
> I meant to add it to the nu_scripts/benchmarks folder as a script.
>
ooh i did not even know / remember that we have this :open_mouth:

**Comment by amtoine** [comment]
thanks @sholderbach for the review :pray: 

just to be sure, before i do anything, **do we move this to the `nu_scripts?**
i'm fine with that :yum:

**Comment by fdncred** [comment]
I'd move it, but if we can make it a valid benchmark, i'd be fine with it staying too.

**Comment by amtoine** [comment]
i think i'll move it to `nu_scripts`, i'd wish i had more time for this but it's a bit hard to run `cargo bench` on my machine :confused:

**Pull request closed without merging**